### PR TITLE
[release/v2.8] Bump rancher/system-agent to v0.3.5-rc1

### PR DESCRIPTION
**Pull request by Oats87**
## Issue: 
https://github.com/rancher/rancher/issues/43658
 
## Problem
During v2prov/CAPR day-2 operations, it is possible that the system-agent is restarted in the middle of applying a plan, which can lead to unexpected results.

## Solution
Add interlocks to the system-agent to prevent it from restarting when the applyinator is active and stops the applyinator from applying if a restart is pending (to prevent a deadlock)
 
### Manual Testing
I performed manual testing in various scenarios to ensure the interlocks were working as expected, see the screenshot for example
Screenshot 2023-12-12 at 2 00 14 PM

### Automated Testing

* Test types added/modified:
    * None

Summary: The existing integration tests test the change

### Regressions Considerations

_TODO_

Existing / newly added automated tests that provide evidence there are no regressions:
* _TODO_

### CODE REVIEW

File: pkg/capr/installer/server_test.go [hunk 1]
```diff
@@ -42,6 +43,11 @@ func TestHandler_ServeHTTP(t *testing.T) {
 			handler := handler{}
 
 			// act
+			err = settings.SystemAgentInstallScript.Set("https://raw.githubusercontent.com/rancher/system-agent/main/install.sh")
```
Comment (jakefhyde) [inline pkg/capr/installer/server_test.go:46]: Should we pick something more stable than `main` for this?

File: scripts/provisioning-tests [hunk 2]
```diff
@@ -61,9 +61,9 @@ mkdir -p /var/lib/rancher/$DIST/agent/images
 grep PodTestImage ./tests/v2prov/defaults/defaults.go | cut -f2 -d'"' > /var/lib/rancher/$DIST/agent/images/pull.txt
 grep MachineProvisionImage ./pkg/settings/setting.go | cut -f4 -d'"' >> /var/lib/rancher/$DIST/agent/images/pull.txt
 mkdir -p /usr/share/rancher/ui/assets
-curl -sLf https://github.com/rancher/system-agent/releases/download/${CATTLE_SYSTEM_AGENT_VERSION}/rancher-system-agent-amd64 -o /usr/share/rancher/ui/assets/rancher-system-agent-amd64
-curl -sLf https://github.com/rancher/system-agent/releases/download/${CATTLE_SYSTEM_AGENT_VERSION}/rancher-system-agent-arm64 -o /usr/share/rancher/ui/assets/rancher-system-agent-arm64
-curl -sLf https://github.com/rancher/system-agent/releases/download/${CATTLE_SYSTEM_AGENT_VERSION}/system-agent-uninstall.sh -o /usr/share/rancher/ui/assets/system-agent-uninstall.sh
+curl -sLf ${CATTLE_SYSTEM_AGENT_DOWNLOAD_PREFIX}/${CATTLE_SYSTEM_AGENT_VERSION}/rancher-system-agent-amd64 -o /usr/share/rancher/ui/assets/rancher-system-agent-amd64
```
Comment (jakefhyde) [inline scripts/provisioning-tests:64]: How does this work if the env variable is only defined in `package/Dockerfile`?

**Review by HarrisonWAffel — COMMENTED** [review]
Other than Jakes comments, lgtm

File: scripts/provisioning-tests (hunk 2 above)
Comment (Oats87) [reply scripts/provisioning-tests:64]: https://github.com/rancher/rancher/blob/930b53bb3779ebaa02b49948877c727c008f8833/scripts/provisioning-tests#L43 this is how

File: pkg/capr/installer/server_test.go (hunk 1 above)
Comment (Oats87) [reply pkg/capr/installer/server_test.go:46]: Without turning it into a burden, I don't think so. 

If we ever move away from the `main` branch having a functional `install.sh` then we can deal with it then. 

the techdebt here would be to move the source of truth for this script somewhere that can be easily accessed

**Review by jakefhyde — DISMISSED** [review]

**Review by HarrisonWAffel — DISMISSED** [review]

**Review by HarrisonWAffel — APPROVED** [review]

**Review by jakefhyde — APPROVED** [review]

**Pull request merged**